### ISSUE TITLE
Can't connect on win11 and MacOS 14-15

### ISSUE THREAD

**Issue by x0777** [jawj/IKEv2-setup#163]
Good afternoon! 
Many thanks for your script. I've been using it for 2 years now. It works perfectly and without any errors on iOS 16-18 and macOS 10 Catalina.

However, there are a couple of issues:

1) It has stopped connecting on macOS 14-15. There are constant unsuccessful reconnection attempts or it just gets stuck at the connection stage.
2) Since a certain Windows 11 update, it has stopped connecting and gives the error "IKE authentication credentials are unacceptable."
Tested on Ubuntu 18.04 and 24.04 servers.

What could be the cause, and where can I fix this?

*Just out of curiosity, I tried installing a similar script from GitHub, and it works on all devices.

**Comment by jawj** [comment]
* Have you tried re-generating the macOS `.mobileconfig` file by installing the latest version of the script? I find VPN in the latest macOS versions a bit flaky, but not nearly as bad as that.
* I will take a look at WIn 11 when I can.

It might be helpful to see the 'similar script', in order to see what configuration differences there are.

**Comment by x0777** [comment]
Yes, I just did a clean install on Ubuntu 24.04 today. I tried to run it on macOS 15.0.1 — same symptoms since macOS 14.

> It might be helpful to see the 'similar script', in order to see what configuration differences there are.

https://github.com/hwdsl2/setup-ipsec-vpn/tree/master - link to "similar script"
https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/ikev2-howto.md#ikev2-troubleshooting - link to troubleshooting page of that script, where i found similar errors and their solutions. Might be helpful as well.

**Comment by x0777** [comment]
Hello! Are there any updates or maybe changes to script coming? Or you didn't encountered and can't replicate same issues described above?

**Comment by ww9rivers** [comment]
Not sure if this helps as I don't have enough information. But I hope it does......

I [had the same error message](https://github.com/strongswan/strongswan/discussions/2575) on a Windows 10 box (I am running strongSwan 5.9.13 on Ubuntu 24.04.1 for the VPN server). The Q&A I had, plus this script, helped.

Thanks to @jawj for this script!

**Comment by QQmberling** [comment]
I had a similar issue after another Windows update. Solved it by removing the client VPN configuration and adding it back following the instructions in the vpn-instructions.txt file.

```
Add-VpnConnection -Name "${VPNHOST}" \`
  -ServerAddress "${VPNHOST}" \`
  -TunnelType IKEv2 \`
  -EncryptionLevel Maximum \`
  -AuthenticationMethod EAP \`
  -RememberCredential

Set-VpnConnectionIPsecConfiguration -ConnectionName "${VPNHOST}" \`
  -AuthenticationTransformConstants GCMAES256 \`
  -CipherTransformConstants GCMAES256 \`
  -EncryptionMethod GCMAES256 \`
  -IntegrityCheckMethod SHA384 \`
  -DHGroup ECP384 \`
  -PfsGroup ECP384 \`
  -Force
```

My guess is that the client encryption proposals have been changed after the Windows update.
